tests/test_expectation_maximization.py

```diff
@@ -107,10 +107,54 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     rots = test_ir.grid_SO3_uniform(n_particles)
     xy_plane = test_ir.generate_xy_plane(n_pix)
 
-    slices, xyz_rotated = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
+    slices, xyz_rotated_planes = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
 
     assert slices.shape == (n_particles, n_pix, n_pix)
-    assert xyz_rotated.shape == (n_particles, 3, n_pix**2)
+    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix**2)
+
+    # ---------------------------------
+    # 90-degree rotation test.
+    # Map has ones in central xz-plane.
+    # Rotating by -90 degrees about y
+    # should produce a slice of ones.
+    rot_test_map = np.zeros((n_pix, n_pix, n_pix))
+    rot_test_map[2] = np.ones((n_pix, n_pix))
+
+    rot_mat = np.array(
+        [
+            [[0, 0, 1], [0, 1, 0], [-1, 0, 0]],
+        ]
+    )
+
+    slices, xyz_rotated_planes = test_ir.generate_slices(
+        rot_test_map, xy_plane, n_pix, rot_mat
+    )
+    assert np.allclose(slices[0], np.ones_like(slices[0]))
+    # ---------------------------------
+
+    # ---------------------------------
+    # 180-degree rotation test.
+    # Map has ones in central xz-plane.
+    # Rotating by 180 degrees about z
+    # should produce a similar matrix,
+    # namely a slice of ones in the -1-line.
+    rot_test_map = np.zeros((4, 4, 4))
+    rot_test_map[2] = np.ones((4, 4))
+
+    rot_mat = np.array(
+        [
+            [[-1, 0, 0], [0, -1, 0], [0, 0, 1]],
+        ]
+    )
+
+    expected_slice = np.zeros((4, 4))
+    expected_slice[:, n_pix - 2 - 1] = 1
```

Comment:
Make n_pix general. You can make it a random integer. It's a good assurance that the test is indeed general.

I think it's

```
rot_test_map = np.zeros((n_pix, n_pix, n_pix))
rot_test_map[n//2] = np.ones((n_pix, n_pix))
#...
expected_slice[:, n_pix//2 - 1] = 1
expected_slice = np.zeros((n_pix, n_pix))
```

Reply:
Good call - oversight on my part I had meant to replace these with n_pix. I'll add randomization to the testing as well.